src/rgw/rgw_rados.h

```diff
@@ -1869,7 +1869,26 @@ class RGWPeriod
   int get_zonegroup(RGWZoneGroup& zonegroup,
 		    const string& zonegroup_id);
 
-  bool is_single_zonegroup(CephContext *cct, RGWRados *store);
+  bool is_single_zonegroup()
+  {
+      return (period_map.zonegroups.size() == 1);
+  }
+
+  /*
+    returns true if there are several zone groups with a least one zone
+   */
+  bool is_multi_zonegroups_with_zones()
+  {
+    int count = 0;
+    for (const auto& zg:  period_map.zonegroups) {
+      if (zg.second.zones.size() > 0) {
+	if (count++ > 0) {
+	  return true;
+	}
+      }
+    }
+    return false;
+  }
```

Comment:
what do you think about making a `get_num_zones()` that just counts the zones from all zonegroups? then you wouldn't need to split single-zonegroup and multi-zonegroup cases

`need_to_log_metadata()` could use `get_num_zones() > 1` and `is_syncing_bucket_meta()` could use `get_num_zones() == 1`